src/blackmisc/aviation/flightplanaircraftinfo.cpp

```diff
@@ -0,0 +1,388 @@
+
+    CFlightPlanAircraftInfo::CFlightPlanAircraftInfo(QString equipmentCodeAndAircraft)
+    {
+        equipmentCodeAndAircraft = equipmentCodeAndAircraft.trimmed().toUpper().replace(" ", "");
+        const int numberSlash = equipmentCodeAndAircraft.count("/");
+        const int numberHypen = equipmentCodeAndAircraft.count("-");
+        if (numberHypen == 1 && numberSlash == 2)
+        {
+            parseIcaoEquipmentCode(equipmentCodeAndAircraft);
+        }
+        else if (numberSlash >= 1 && numberSlash <= 2 && numberHypen == 0)
+        {
+            parseFaaEquipmentCode(equipmentCodeAndAircraft);
+        }
+        else
+        {
+            parseUnknownEquipmentCode(equipmentCodeAndAircraft);
+        }
+    }
+
+    QString CFlightPlanAircraftInfo::asIcaoString() const
+    {
+        // Avoid returning empty wake turbulence categories and send it to the server.
+        // This can in particular happen when translating from FAA codes to ICAO codes.
+        // This sets a default  wake turbulence category of MEDIUM if the category is unknown otherwise.
+        QChar wtc;
+        if (!m_wtc.isUnknown())
+        {
+            wtc = m_wtc.toQString().at(0);
+        }
+        else
+        {
+            wtc = CWakeTurbulenceCategory(CWakeTurbulenceCategory::MEDIUM).toQString().at(0);
+        }
+
+        return m_aircraftIcao.getDesignator() % "/" % wtc % "-" % m_comNavEquipment.toQString() % "/" % m_ssrEquipment.toQString();
+    }
+
+    QString CFlightPlanAircraftInfo::asFaaString() const
+    {
+        // TCAS prefix (T/) is not used
+        QString s;
+        if (m_wtc.isCategory(CWakeTurbulenceCategory::HEAVY))
+        {
+            s = "H/";
+        }
+        else if (m_wtc.isCategory(CWakeTurbulenceCategory::SUPER))
+        {
+            s = "J/";
+        }
+        s += m_aircraftIcao.getDesignator() % "/" % equipmentToFaaCode(m_comNavEquipment, m_ssrEquipment);
+        return s;
+    }
+
+    CAircraftIcaoCode CFlightPlanAircraftInfo::getAircraftIcao() const
+    {
+        return m_aircraftIcao;
+    }
+
+    CComNavEquipment CFlightPlanAircraftInfo::getComNavEquipment() const
+    {
+        return m_comNavEquipment;
+    }
+
+    CSsrEquipment CFlightPlanAircraftInfo::getSsrEquipment() const
+    {
+        return m_ssrEquipment;
+    }
+
+    CWakeTurbulenceCategory CFlightPlanAircraftInfo::getWtc() const
+    {
+        return m_wtc;
+    }
+
+    QString CFlightPlanAircraftInfo::convertToQString(bool) const
+    {
+        return asIcaoString();
+    }
+
+    void CFlightPlanAircraftInfo::parseIcaoEquipmentCode(const QString &equipment)
+    {
+        // Example: B789/H-SDE1E2E3FGHIJ2J3J4J5M1RWXY/LB1D1
+        QStringList firstSplit = equipment.split("/");
+        Q_ASSERT_X(firstSplit.size() == 3, Q_FUNC_INFO, "Cannot split string as required for the ICAO format");
```

Comment:
- [ ] Is it correct that this should be an assert? Do we have a separate function for validating input?